lib/aruba/platform.rb

```diff
@@ -1,257 +1,29 @@
-      ::File.join(RbConfig::CONFIG['bindir'], RbConfig::CONFIG['ruby_install_name'])
-    end
-
-    # @deprecated
-    # Add newline at the end
-    def ensure_newline(str)
-      deprecated('The use of "#ensure_newline" is deprecated. It will be removed soon')
-
-      str.chomp << "\n"
-    end
-
-    def require_matching_files(pattern, base)
-      if RUBY_VERSION < '1.9.3'
-        ::Dir.glob(::File.expand_path(pattern, base)).each { |f| require File.join(File.dirname(f), File.basename(f, '.rb')) }
-      else
-        ::Dir.glob(::File.expand_path(pattern, base)).each { |f| require_relative f }
-      end
-    end
-
-    # Create directory and subdirectories
-    def mkdir(dir_name)
-      dir_name = ::File.expand_path(dir_name)
-
-      ::FileUtils.mkdir_p(dir_name) unless ::File.directory?(dir_name)
-    end
-
-    # Remove file, directory + sub-directories
-    def rm(paths, options = {})
-      paths = Array(paths).map { |p| ::File.expand_path(p) }
-
-      FileUtils.rm_r(paths, options)
-    end
-
-    # Get current working directory
-    def getwd
-      Dir.getwd
-    end
-
-    # Change to directory
-    def chdir(dir_name, &block)
-      dir_name = ::File.expand_path(dir_name.to_s)
-
-      begin
-        if RUBY_VERSION <= '1.9.3'
-          old_env = ENV.to_hash.dup
-        else
-          old_env = ENV.to_h.dup
-        end
-
-        ENV['OLDPWD'] = getwd
-        ENV['PWD'] = dir_name
-        ::Dir.chdir(dir_name, &block)
-      ensure
-        ENV.clear
-        ENV.update old_env
-      end
-    end
-
-    # Touch file, directory
-    def touch(args, options)
-      FileUtils.touch(args, options)
-    end
-
-    # Copy file/directory
-    def cp(args, options)
-      FileUtils.cp_r(args, options)
-    end
-
-    # Change mode of file/directory
-    def chmod(mode, args, options)
-      FileUtils.chmod_R(mode, args, options)
-    end
-
-    # Exists and is file
-    def file?(f)
-      File.file? f
-    end
-
-    # Exists and is directory
-    def directory?(f)
-      File.directory? f
-    end
-
-    # Path Exists
-    def exist?(f)
-      File.exist? f
-    end
-
-    # Path is executable
-    def executable_file?(f)
-      File.file?(f) && File.executable?(f)
-    end
-
-    # Expand path
-    def expand_path(path, base)
-      File.expand_path(path, base)
-    end
-
-    def absolute_path?(path)
-      Pathname.new(path).absolute?
-    end
-
-    # Check if command is relative
-    #
-    # @return [TrueClass, FalseClass]
-    #   true
-    #     * bin/command.sh
-    #
-    #   false
-    #     * /bin/command.sh
-    #     * command.sh
-    def relative_command?(path)
-      p = Pathname.new(path)
-      p.relative? && p.basename != p
-    end
-
-    # Write to file
-    def write_file(path, content)
-      if RUBY_VERSION < '1.9.3'
-        File.open(path, 'wb') do |f|
-          f.print content
-        end
-      else
-        File.write(path, content)
-      end
-    end
+  PLATFORM_MUTEX = Mutex.new
```

Comment:
Paranoia... I read something about class instance variables being not thread safe.